modules/web/client/src/main/scala/observe/ui/components/services/ObservationSyncer.scala

```diff
@@ -64,8 +68,27 @@ object ObservationSyncer:
                   props.nighttimeObservation.async.mod(_.map(_.withConfig(config)))
 
             (fetchSummary, fetchSequence).parTupled >>= ((_, configEither) =>
-              configEither.toOption // TODO Set breakpoints from ODB
-                .map(config => sequenceApi.loadObservation(obsId, config.instrument))
+              configEither.toOption
+                .map: config =>
+                  def getBreakPoints(sequence: Option[ExecutionSequence[?]]): Set[Step.Id] =
+                    sequence
+                      .map(s => s.nextAtom +: s.possibleFuture)
+                      .orEmpty
+                      .flatMap(_.steps.toList)
+                      .collect { case s if s.breakpoint === Breakpoint.Enabled => s.id }
+                      .toSet
+
+                  val initialBreakpoints: Set[Step.Id] =
+                    config match
+                      case InstrumentExecutionConfig.GmosNorth(executionConfig) =>
+                        getBreakPoints(executionConfig.acquisition) ++
+                          getBreakPoints(executionConfig.science)
+                      case InstrumentExecutionConfig.GmosSouth(executionConfig) =>
+                        getBreakPoints(executionConfig.acquisition) ++
+                          getBreakPoints(executionConfig.science)
+
+                  sequenceApi.loadObservation(obsId, config.instrument) >>
+                    sequenceApi.setBreakpoints(obsId, initialBreakpoints.toList, Breakpoint.Enabled)
```

Comment:
So we are submitting breakpoints from the odb on load? I think we are doing thay already on the odb

Reply:
Yes. This sets the initial breakpoints for the portion of the sequence that the UI reads (max 100 atoms). Does the server read all that?

Reply:
We only read one atom, thus this call will set the breakpoint for the first atom and ignore the rest

Reply:
Can we keep the set of breakpoints in the server even if it hasn't read the atoms yet? Otherwise we would disable setting breakpoints beyond the current atom in the UI.